pallets/asset-manager/src/lib.rs

```diff
@@ -232,6 +233,28 @@ pub mod pallet {
 			Self::deposit_event(Event::AssetTypeChanged(asset_id, new_asset_type));
 			Ok(())
 		}
+
+		/// Remove a given aassetId -> assetType association
+		#[pallet::weight(T::WeightInfo::remove_existing_asset_type())]
+		pub fn remove_existing_asset_type(
+			origin: OriginFor<T>,
+			asset_id: T::AssetId,
+		) -> DispatchResult {
+			T::AssetModifierOrigin::ensure_origin(origin)?;
+
+			let asset_type =
+				AssetIdType::<T>::get(&asset_id).ok_or(Error::<T>::AssetDoesNotExist)?;
+
+			// Remove from AssetIdType
+			AssetIdType::<T>::remove(&asset_id);
+			// Remove from AssetTypeId
+			AssetTypeId::<T>::remove(&asset_type);
+			// Remove previous asset type units per second
+			AssetTypeUnitsPerSecond::<T>::remove(&asset_type);
```

Comment:
Why do we have two extrinsics for adding an asset and only one for removing? Is there a use case for having an asset registered without units-per-second also being registered?

Reply:
There might be cases where we want to register an asset, but where we dont want to set the units per second because we dont want to support this asset for XCM fee payment.

Reply:
I will check whether it exists before I remove the units per second, thanks!

Reply:
I finally made a test ensuring it does not panic when we try to remove AssetTypeUnitsPerSecond even if the key does not exist